Project-19-React-router/frontend/src/util/auth.js

```diff
@@ -0,0 +1,40 @@
+import { redirect } from "react-router-dom";
+
+export function getTokenDuration() {
+  const storedExpirationDate = localStorage.getItem("expiration");
+  const expirationDate = new Date(storedExpirationDate);
+  const now = new Date();
+  const duration = expirationDate.getTime() - now.getTime();
+  return duration;
+}
+
+export function getAuthToken() {
+  const token = localStorage.getItem("token");
+
+  if (!token) {
+    return null;
+  }
+
+  const tokenDuration = getTokenDuration();
+
+  if (tokenDuration < 0) {
+    return "EXPIRED";
+  }
+
+  return token;
+}
+
+export function tokenLoader() {
+  const token = getAuthToken();
+  return token;
+}
```

Comment:
@sknepali The tokenLoader function is redundant as it simply calls and returns the result of getAuthToken. We can remove this function.